London | May-2025| Merhawit Teages | Form Controls

merhawit-teages opened the pull request:
<!--

You must title your PR like this:

REGION | COHORT_NAME | FIRST_NAME LAST_NAME | PROJ_NAME

For example,

London | May-2025 | Carol Owen | Wireframe

Complete the task list below this message.
If your PR is rejected, check the task list.

-->

## Learners, PR Template

Self checklist
- [x] I have committed my files one by one, on purpose, and for a reason
- [x] I have titled my PR with London | May-2025| Merhawit Teages | Form Controls
- [x] I have tested my changes
- [x]  My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/reviewing/style-guide/)
- [x] My changes meet the [requirements](./README.md) of this task

## Changelist
I created a semantic HTML form for a t-shirt ordering page.  
The form collects:
Full name
Email
T-shirt colour (3 options)
T-shirt size (6 options)

Briefly explain your PR.
I also:
Connected the form to `results.html` using `method="get"` to show submitted data
Ensured all inputs have associated labels

## Questions
Is there anything I could improve in terms of accessibility?
Is the way I used the `results.html` page with query strings okay for this project?

cjyuan commented on Form-Controls/index.html:
```diff
@@ -8,20 +8,54 @@
+        <label for="name">Full Name:</label>
+        <input type="text" id="name" name="name" required minlength="2" />
+        <br><br>
+        <!-- Customer's password -->
+        <label for="password">Password:</label>
+        <input type="password" id="password" name="password" required minlength="8" />
+        <br><br>
```
It’s best to avoid adding features that aren’t specified in the requirements. If you believe a feature would be valuable, a better practice is to request for approval beforehand.

cjyuan commented on Form-Controls/index.html:
```diff
@@ -8,20 +8,54 @@
-        this will also help you fill in your PR message later-->
-      </form>
+      <form action="results.html" method="get">
+        <!--Customer's name -->
+        <label for="name">Full Name:</label>
+        <input type="text" id="name" name="name" required minlength="2" />
```
Currently a user can enter a name consisting of only space characters (e.g., " "). Can you enforce a stricter validation rule using the `pattern` attribute to disallow any name that contains only space characters?

cjyuan reviewed (commented):
1. According to W3C Markup Validation Service, there are errors in the HTML code. Can you fix them?

2. This PR branch is not considered "clean" because it contains modified files unrelated to Form-Controls implementation. The unrelated files are those in the "Wireframe" folder. Can you restore the files in the "Wireframe" frolder to their original state (so that they are identical to the same files in the `main` branch or in CYF's repo)?

3. There is a way to check a checkbox in the PR description without using emoji character. I have changed one of them for you in the PR description. Can you update the other four checkboxes accordingly? 

---
A well-prepared PR makes it easier for reviewers to approve it with minimal back-and-forth.
The "PR Essentials" file on the [Canvas](https://codeyourfutur-yov6609.slack.com/archives/C07QX99JK7B) in the #cyf-code-review channel has some helpful tips to make your PRs more robust and ready for review.
Can you take a few minutes to check against the essentials, including how to prepare PR description? Doing so can help speed up the review process.

And as a practice to get better at using AI tools, try running your code by ChatGPT. You can ask ChatGPT
```
review HTML code:
...        <--- paste your HTML code here
```

Then take a look at the suggestions it gives you, and pick out the ones that actually make sense for your project. It's a great way to learn and improve your work.

merhawit-teages commented:
I’ve now fixed the HTML validation issues in the form and cleaned up the pull request so that it only includes files related to the Form Controls implementation. I removed the unrelated files from the Wireframe folder to keep this branch clean and focused. I’ve also updated the checkboxes in the PR description as requested. Please let me know if there’s anything else I should change.

cjyuan reviewed (commented):
Changes made to the code looks great!

**Can you restore the files in the "Wireframe" folder to their original state instead of deleting them?**

This PR was created from your `feature/Form--controls` branch aginst CodeYourFutre's `main` branch. As as result, under the "Files changed" tab, Github will show the **difference** between your branch and CYF's `main`. Please note that deleted files are also considered as "changes". 
![image](https://github.com/user-attachments/assets/1bfb9074-44ba-47cf-9020-8528918797d7)

What you need to do to revert the changes made in the "Wireframe" folder is to replace those files by the same files from your `main` branch: https://github.com/merhawit-teages/Module-Onboarding/tree/main/Wireframe

Note: The files in CYF's `main` and the files in your `main` are identical.

---

To check the boxes in the PR description, the Markdown syntax to prepare checked box is
```
- [x] ...
```
not
```
[x] ...
```

merhawit-teages commented:
I’ve now restored the Wireframe folder files to match the main branch as requested, and kept only the intended changes in the Form-Controls folder.

Please let me know if there’s anything else I should adjust.

merhawit-teages commented:
I’ve now restored all the files in the Wireframe folder to match the main branch and updated the results.html page to display the submitted form data dynamically correctly.
Please let me know if there’s anything else I should adjust.

cjyuan commented:
Well done!
